terraform/resources/langfuse.tf

```diff
@@ -0,0 +1,62 @@
+# S3 Bucket for current environment
+resource "aws_s3_bucket" "langfuse" {
+  bucket = "langfuse-${var.environment}-bucket"
+}
+
+# IAM user for Langfuse
+resource "aws_iam_user" "langfuse" {
+  name = "langfuse-${var.environment}-s3-user"
+}
+
+# Bucket policy
+resource "aws_iam_user_policy" "langfuse" {
+  name = "langfuse-${var.environment}-policy"
+  user = aws_iam_user.langfuse.name
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject",
+          "s3:ListBucket"
+        ]
+        Resource = [
+          aws_s3_bucket.langfuse.arn,
+          "${aws_s3_bucket.langfuse.arn}/*"
+        ]
+      }
+    ]
+  })
+}
+
+# Access key for the IAM user
+resource "aws_iam_access_key" "langfuse" {
+  user = aws_iam_user.langfuse.name
+}
+
+# Kubernetes secret for Langfuse S3 credentials
+resource "kubernetes_secret" "langfuse_s3_credentials" {
+  metadata {
+    name      = "langfuse-s3-credentials"
+    namespace = "default"
```

Comment:
this probably needs to be in the same namespace as the langfuse pod referencing the secret -- so production or staging

Reply:
@sunu since we're now deploying prod and staging on different clusters, we deploy the main zeno chart (with langfuse as dependency) into the default namespace. So I think this is correct the way it is.